sdk/core/Azure.Core/src/Internal/ArrayBackedPropertyBag.cs

```diff
@@ -260,35 +258,51 @@ private int GetIndex(TKey key)
             if (_count > 1 && _second.Key.Equals(key))
                 return 1;
 
+            if (_count <= 2)
+                return -1;
+
+            Kvp[] rest = GetRest();
             int max = _count - 2;
             for (var i = 0; i < max; i++)
             {
-                if (_rest[i].Key.Equals(key))
+                if (rest[i].Key.Equals(key))
                     return i + 2;
             }
             return -1;
         }
 
         internal void Dispose()
         {
-            CheckDisposed();
 #if DEBUG
+            if (_disposed)
```

Comment:
Why is this Debug only?

Reply:
This field is used to catch the cases when `ArrayBackedPropertyBag` is used after disposing, but I don't want to crash user app if it happens cause with current implementation it is safe to use instance after disposing.

Reply:
Also, that is a field that we can omit to make the struct smaller.